StockAnalyzerService/StockAnalyzerService/Service/HttpCalls.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace StockAnalyzerService.Service {
+    public class HttpCalls: IHttpCalls {
+
+        private readonly ILogger _logger;
+        public HttpCalls(ILogger<HttpCalls> logger) {
+            _logger = logger;
+        }
+
+        public async Task<T> Get<T>(HttpClient httpClient, string urlParameters) {
+            try {
+                string url = httpClient.BaseAddress + urlParameters;
+                HttpResponseMessage response = await httpClient.GetAsync(url);
+                if (response.IsSuccessStatusCode) {
+                    return await response.Content.ReadAsAsync<T>();
+                }
+                return default;
+            } catch (Exception ex) {
+                _logger.LogError(ex, "Error when making a http GET call");
```

Comment:
Not super important right now because we will revisit exception handling later once the code is developed. Wanted to make a note, can we add information about what httpClient and urlParameter caused the error?

Reply:
I can look into it, but I'm not sure. I think it depends on what the exception actually says

Reply:
You can always record the variables. If we want to use them to figure out the issue, then we have the information available to us